crapssim/bet.py

```diff
@@ -826,6 +1041,120 @@ class AnyCraps(_SimpleBet):
     payout_ratio: int = 7
 
 
+class Horn(_WinningLosingNumbersBet):
+    """One-roll bet split across 2, 3, 11, and 12; loses on all other totals."""
+
+    winning_numbers: list[int] = [2, 3, 11, 12]
+    losing_numbers: list[int] = list(ALL_DICE_NUMBERS - {2, 3, 11, 12})
+
+    def __init__(self, amount: SupportsFloat) -> None:
+        super().__init__(amount)
+
+    def get_winning_numbers(self, table: "Table") -> list[int]:
+        return self.winning_numbers
+
+    def get_losing_numbers(self, table: "Table") -> list[int]:
+        return self.losing_numbers
+
+    def get_payout_ratio(self, table: "Table") -> float:
+        total = table.dice.total
+        if total in (2, 12):
+            return 6.75
+        if total in (3, 11):
+            return 3.0
+        raise NotImplementedError
+
+    def copy(self) -> "Horn":
+        return self.__class__(self.amount)
+
+    @property
+    def _placed_key(self) -> Hashable:
+        return type(self)
```

Comment:
These should be inherited by the Bet class, so no need to re-define here. Same comment for World, Big6, Big8 bets